framework/utils/xmlUtils.py

```diff
@@ -131,36 +133,100 @@ def newNode(tag,text='',attrib={}):
   el.text = fixXmlText(text)
   return el
 
-def newTree(name,attrib={}):
+def newTree(name,attrib=None):
   """
     Creates a new tree with named node as its root
     @ In, name, string, name of root node
     @ In, attrib, dict, optional, attributes for root node
     @ Out, tree, xml.etree.ElementTree.ElementTree, tree
   """
+  if attrib is None:
+    attrib = {}
   name = fixXmlTag(name)
   tree = ET.ElementTree(element=newNode(name))
   tree.getroot().attrib = dict(attrib)
   return tree
 
+def fixTagsInXpath(_path):
+  """
+    Fixes tags/attributes/text in an xpath string to use allowable characters
+    @ In, _path, str, xpath string
+    @ Out, out, str, modified string
+  """
+  # XPATH OPTIONS:
+  # tag wildcards: * . // ..
+  # modifiers
+  #  [@attrib]
+  #  [@attrib='value']
+  #  [tag]
+  #  [tag='text']
+  #  [position]  --> same as [tag] for this purpose
```

Comment:
A more detailed description will benefit the future developers. For my understanding, these are the supported XPATH syntax by the python ```xml.etree.ElementTree```. Basically, we can use these syntax to extract the data from the given XML file. Please correct me if I'm wrong. If this is the case, I think we should document this in the manual, and add tests to test this feature. To me, this will make RavenOutput PP more powerful.

Reply:
All of the possibilities are covered in unit tests, and I don't have a use case for a regression test past what we already have as tests.

I don't think there's a good reason to replicate the documentation for XPath and ElementTree here; the manual does specifically link the documentation that explains the Python implementation that we take advantage of.  This comment here should not be seen as a full briefing of the capabilities; rather, it is a bullet list of the cases we need to cover when guarding tags, attributes, and text against illegal XML characters.  The list is meant to inform future developers of the structure of the code, not teach them XPath syntax.

Reply:
I see your point. This is unclear to me at the first glance. I do check the XPath and I could not found all the options. When I check the python ```xml.etree.ElementTree```, I realized that this just he supported XPATH syntax. Maybe you can change a little bit about the description to make it more clear. Other than that, I do not have any objections.

Reply:
In `findPath` docstrings I do link directly to the etree documentation where it talks about xpath syntax, where you actually submit a search.  `fixTagsInXpath` is a utility just to protect against illegal characters consistently; I'm not clear on what description is unclear.  Which do you want to see changed?

Reply:
Okay, I clarified.

Reply:
Thanks.